app/screens/channel_info/notification_preference/index.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {getMyCurrentChannelMembership} from '@mm-redux/selectors/entities/channels';
+
+import NotificationPreference from './notification_preference';
+
+function mapStateToProps(state) {
+    const channelMember = getMyCurrentChannelMembership(state);
+
+    return {
+        channelId: channelMember && channelMember.channel_id,
+        userId: channelMember && channelMember.user_id,
+        notifyProps: channelMember && channelMember.notify_props,
```

Comment:
you could write it like this:

```suggestion
    return {
        channelId: channelMember?.channel_id,
        userId: channelMember?.user_id,
        notifyProps: channelMember?.notify_props,
```

Also not sure why you need to check if `channelMember` is defined but in the case it could be undefined or null, meaning that channelId, userId & notifyProps won't be set, maybe the component needs to render `null` instead of `ChannelInfoRow` if any of those values are not set. thoughts?